packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,168 @@
+}
+
+@Component({
+  tag: 'va-file-input',
+  styleUrl: 'va-file-input.css',
+  shadow: true,
+})
+
+export class VaFileInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the button input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button
+   */
+  @Prop() buttonText: string;
+
+  /**
+   * Set the input to required and render the (Required) text.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * Allow the input to accept multiple files.
+   */
+  @Prop() multiple?: boolean = false;
+
+  /**
+   * A comma-separated list of unique file type specifiers.
+   */
+  @Prop() accept?: string;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Emit component-library-analytics events on the file input change event.
+   */
+   @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * The event emitted when the file input value changes.
+   */
+  @Event() vaChange: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when the
+   * file input changes and enableAnalytics is true.
+   */
+   @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.vaChange.emit({files: target.files});
+    /**
+     * Clear the original input, otherwise events will be triggered
+     * with empty file arrays and sometimes uploading a file twice will
+     * not work.
+     */
+    target.value = null;
+
+    if (this.enableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-file-input',
+        action: 'change',
+        details: {
+          label: this.label
+        },
+      });
+    }
+  };
+
+  private handleButtonClick = () => { 
+    this.el.shadowRoot.getElementById('fileInputField').click();
+  }
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+  }
+
+  disconnectedCallback() {
+    i18next.off('languageChanged');
+  }
+
+  render() {
+    const { 
+      label,
+      name, 
+      buttonText, 
+      required, 
+      multiple,
+      accept,
+      error,
+    } = this;
+    return (
+      <Host>
+        {label && (
+          <label htmlFor="fileInputButton">
+            {label}
+            {required && <span class="required">{i18next.t('required')}</span>}
+          </label>
+        )}
+        <slot></slot>
+        <span id="error-message" role="alert">
+          {error && (
+            <Fragment>
+              <span class="sr-only">{i18next.t('error')}</span> 
+              {error}
+            </Fragment>
+          )}
+        </span>
+        <va-button
+          id="fileInputButton"
+          aria-label={i18next.t(label)}
```

Comment:
I don't think we should pass a prop like this to i18next because we don't know if we're going to have a translation for it.

Reply:
@bkjohnson I might have just misunderstood how the translations will map. If there is no translation available, would it default to the provided label text (english)? Alternately, if there is no label, would it just not populate with anything? Is there any harm in setting it up as "translation-ready"?

Reply:
The `t()` function takes a key as its first argument and uses it to look up values in a [language-specific translation object like this](https://github.com/department-of-veterans-affairs/component-library/blob/3fc3b7d33d9cfe5ac51a54d7aed0b445c26cfe05/packages/core/src/i18n/translations/en.js#L1-L26). For template strings I've seen it return `undefined` if the value isn't there, so I'm not totally sure why we still see text at all when we're in Spanish mode.

As far as making something translation-ready, I asked about [something similar in the past](https://github.com/department-of-veterans-affairs/component-library/pull/438#discussion_r888220759), and @k80bowman has said that [we should only provide translations for a component if we can get all of the values](https://github.com/department-of-veterans-affairs/component-library/pull/448#discussion_r891584148). However, that was said before we had linting for string literals, so now there is less of a chance of us forgetting to translate a piece of text, but partially translated components still might not be a good UX :shrug:.

Reply:
That makes sense and thanks! I'll make some updates here.

Reply:
@bkjohnson I have removed `i18next` so I think that will resolve the translation concerns now.  https://github.com/department-of-veterans-affairs/component-library/pull/507/commits/1d8ec3cec7cde235e66aebd083f4e07dc98c29f0

Reply:
Yeah, I'm not a fan of partially translated components if we can avoid it. There may be times we can't (say the component has already been translated, but we add a new error message), but if we don't have any translations yet I'd rather not enable i18next.